File: popup-frameWork/Popup/Dtos/PopupHideResponseDto.cs
using System;

namespace Popup.Dtos
{
    /*
     * 팝업 숨김 처리 후
     * Java 서버가 반환하는 결과 DTO다.
     */
    public class PopupHideResponseDto
    {
        /*
         * 숨김 처리한 사용자 ID
         */
        public string UserId { get; set; } =
            string.Empty;

        /*
         * 숨김 처리한 팝업 ID
         */
        public string PopupId { get; set; } =
            string.Empty;

        /*
         * 서버에 저장된 숨김 유형
         *
         * 현재는 DAYS 값을 사용한다.
         */
        public string HideType { get; set; } =
            string.Empty;

        /*
         * 실제 숨김 종료 일시
         */
        public DateTimeOffset HiddenUntil { get; set; }
    }
}

File: popup-api/src/main/java/com/oksy/popup/dto/PopupHideResponseDto.java
package com.oksy.popup.dto;

import java.time.OffsetDateTime;

/*
 * 사용자 팝업 숨김 처리가 완료된 후
 * Java 서버가 WPF에 반환하는 응답 DTO다.
 */
public record PopupHideResponseDto(

        /*
         * 숨김 처리한 사용자 ID다.
         */
        String userId,

        /*
         * 숨김 처리한 팝업 ID다.
         */
        String popupId,

        /*
         * 숨김 방식이다.
         *
         * 현재 API에서는 UNTIL을 반환한다.
         */
        String hideType,

        /*
         * 팝업 숨김이 끝나는 일시다.
         */
        OffsetDateTime hiddenUntil

) {
}